app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt

```diff
@@ -368,6 +378,94 @@ constructor(
         view?.shareCourse(course)
     }
 
+    /**
+     * User course operations
+     */
+
+    fun toggleFavorite() {
```

Comment:
toggleFavorite & toggleArchive looks like code duplication, let make a single method for this and will pass `UserCourseAction` as argument